src/tendencies/non_orographic_gravity_wave.jl

```diff
@@ -326,33 +333,38 @@ function non_orographic_gravity_wave_forcing(
                     end #(c_hat == 0.0)
                 end # mask = 0
 
-            end # phase speed loop
-
-            # TODO: GFDL option to dump remaining flux at the top of the model
+            end # nc: phase speed loop
 
             # compute the gravity wave momentum flux forcing
             # obtained across the entire wave spectrum at this level.
             if k > source_level
                 rbh = sqrt(ᶜρ[k] * ᶜρ[k - 1])
-                wave_forcing[k] = (ᶜρ[source_level] / rbh) * fm * eps / ᶜdz[k]
-                wave_forcing[k - 1] =
-                    0.5 * (wave_forcing[k - 1] + wave_forcing[k])
+                wave_forcing[k] =
+                    (ᶜρ[source_level] / rbh) * fm * eps / (ᶜz[k] - ᶜz[k - 1])
+                if k == length(ᶜu)
+                    wave_forcing[k - 1] = 0.5 * wave_forcing[k - 1]
+                else
+                    wave_forcing[k - 1] =
+                        0.5 * (wave_forcing[k - 1] + wave_forcing[k])
+                end
             else
-                wave_forcing[k] = 0.0
+                wave_forcing[source_level] = 0.0
```

Comment:
This line can be moved out of the k loop?

Reply:
Agreed!